### PR TITLE
Add application logic for attaching annex to binding contract

### PR DESCRIPTION
**Pull request by meaboutsoftware**
This PR contains changes required to call attach annex in application layer. 

Changes:

- Rename namespace of add annex to attach annex everywhere
- Rename namespace of attach annext to attach annex to binding contract
- Add command and handler for attach annex to binding contract
- Adjust business rule for attaching annex to active binding contract - instead of passing datetime now, we take into consideration from when the annex is valid :)

### CODE REVIEW

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/AttachAnnexToBindingContract/BusinessRules/AnnexCanOnlyBeAttachedToActiveBindingContractRule.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        DateTimeOffset? bindingContractTerminatedAt,
+        DateTimeOffset bindingContractExpiringAt,
+        DateTimeOffset annexValidFrom)
+    {
+        _bindingContractTerminatedAt = bindingContractTerminatedAt;
+        _bindingContractExpiringAt = bindingContractExpiringAt;
+        _annexValidFrom = annexValidFrom;
+    }
+
+    public bool IsMet() => !_bindingContractTerminatedAt.HasValue && _bindingContractExpiringAt > _annexValidFrom;
```
Comment (kamilbaczek) [inline Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/AttachAnnexToBindingContract/BusinessRules/AnnexCanOnlyBeAttachedToActiveBindingContractRule.cs:21]: For me this logic match to rule

!_bindingContractTerminatedAt.HasValue && _bindingContractExpiringAt.


This logic is different rule: Annex Validation Date Should Be In Contract LifeTime (or something like that)
_bindingContractExpiringAt > _annexValidFrom 

What you think?

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core.UnitTests/AttachAnnexToBindingContract/BusinessRules/AnnexCanOnlyBeAttachedToActiveBindingContractRuleTests.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                    bindingContractTerminatedAt,
+                    bindingContractExpiringAt,
+                    now)));
+    }
+
+    [Fact]
+    internal void Given_attach_annex_When_binding_contract_expired_Then_it_is_not_possible_to_attach()
+    {
+        // Arrange
+        var now = DateTimeOffset.UtcNow;
```
Comment (kamilbaczek) [inline Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core.UnitTests/AttachAnnexToBindingContract/BusinessRules/AnnexCanOnlyBeAttachedToActiveBindingContractRuleTests.cs:29]: maybe we can set as private readonly field. To not duplicate in each test

**Review by kamilbaczek — CHANGES_REQUESTED** [review]

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Application/AttachAnnexToBindingContract/AttachAnnexToBindingContractCommandHandler.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+using EvolutionaryArchitecture.Fitnet.Common.Api.ErrorHandling;
+
+[UsedImplicitly]
+internal sealed class AttachAnnexToBindingContractCommandHandler(
+    IBindingContractsRepository bindingContractsRepository) : IRequestHandler<AttachAnnexToBindingContractCommand>
+{
+    public async Task Handle(AttachAnnexToBindingContractCommand command, CancellationToken cancellationToken)
+    {
+        var contract =
```
[the remark this answers is not in the document]
Comment (meaboutsoftware) [reply Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Application/AttachAnnexToBindingContract/AttachAnnexToBindingContractCommandHandler.cs:11]: Done

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core.UnitTests/AttachAnnexToBindingContract/BusinessRules/AnnexCanOnlyBeAttachedToActiveBindingContractRuleTests.cs (hunk 2 above)
Comment (meaboutsoftware) [reply Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core.UnitTests/AttachAnnexToBindingContract/BusinessRules/AnnexCanOnlyBeAttachedToActiveBindingContractRuleTests.cs:29]: Done

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/AttachAnnexToBindingContract/BusinessRules/AnnexCanOnlyBeAttachedToActiveBindingContractRule.cs (hunk 1 above)
Comment (meaboutsoftware) [reply Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/AttachAnnexToBindingContract/BusinessRules/AnnexCanOnlyBeAttachedToActiveBindingContractRule.cs]: Done in a bit different way

Comment (meaboutsoftware) [reply Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/AttachAnnexToBindingContract/BusinessRules/AnnexCanOnlyBeAttachedToActiveBindingContractRule.cs]: @kamilbaczek any feedback on new version?

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Application/AttachAnnexToBindingContract/AttachAnnexToBindingContractCommandHandler.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
+﻿namespace EvolutionaryArchitecture.Fitnet.Contracts.Application.AttachAnnexToBindingContract;
+
+using EvolutionaryArchitecture.Fitnet.Common.Api.ErrorHandling;
+
+[UsedImplicitly]
+internal sealed class AttachAnnexToBindingContractCommandHandler(
```
Comment (kamilbaczek) [inline Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Application/AttachAnnexToBindingContract/AttachAnnexToBindingContractCommandHandler.cs:6]: What you think about returing annex id. To build better rest API. We can do it in next MR

**Review by kamilbaczek — APPROVED** [review]
Good job! Thanks

File: Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/AttachAnnexToBindingContract/BusinessRules/AnnexCanOnlyBeAttachedToActiveBindingContractRule.cs (hunk 1 above)
Comment (kamilbaczek) [reply Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/AttachAnnexToBindingContract/BusinessRules/AnnexCanOnlyBeAttachedToActiveBindingContractRule.cs]: Looks very well for me! Thanks for this change

**Pull request merged**